algebra-testing/src/main/java/com/hubspot/assertj/algebra/error/ResultShouldBeOkWithValue.java

```diff
@@ -12,7 +12,8 @@ private ResultShouldBeOkWithValue(String format, Object... arguments) {
 
   public static <T, E> ErrorMessageFactory shouldBeOkWithValue(Result<T, E> actual, Object value) {
     if (actual.isOk()) {
-      return new ResultShouldBeOkWithValue("Expecting Result to be Ok containing <%s> but contained <%s>", value, actual.unwrapOrElseThrow());
+      T actualValue = actual.unwrapOrElseThrow();
+      return new ResultShouldBeOkWithValue("Expecting Result to be Ok containing <%s> but contained <%s>. expected:<%s> but was:<%s>", value, actualValue, value, actualValue);
     } else {
       return new ResultShouldBeOkWithValue("Expecting Result to be Ok containing <%s> but was Err containing <%s>", value, actual.unwrapErrOrElseThrow());
```

Comment:
No need to do the same for this case imo. If the actual Result is `err` while we're expecting an `ok` result, the two objects to compare are two different things conceptually.